benchmark/primitives/primitives_test.go

```diff
@@ -188,6 +199,83 @@ func BenchmarkMutexWithoutDefer(b *testing.B) {
 	}
 }
 
+func BenchmarkAtomicAddInt64(b *testing.B) {
+	var c int64
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		atomic.AddInt64(&c, 1)
+	}
+	b.StopTimer()
+	if c != int64(b.N) {
+		b.Fatal("error")
+	}
+}
+
+func BenchmarkAtomicTimeValueStore(b *testing.B) {
+	var c atomic.Value
+	t := time.Now()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		c.Store(t)
+	}
+	b.StopTimer()
+}
+
+func atomicStore(a int) {
+	var wg sync.WaitGroup
+	var c atomic.Value
+	for i := 0; i < a; i++ {
+		wg.Add(1)
+		go func() {
+			c.Store(a)
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+}
+
+func mutexStore(a int) {
+	var wg sync.WaitGroup
+	mu := sync.Mutex{}
+	var c int
+	for i := 0; i < a; i++ {
+		wg.Add(1)
+		go func() {
+			mu.Lock()
+			c = a
+			mu.Unlock()
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+}
+
+func BenchmarkValueStoreWithContention(b *testing.B) {
+	benchmarks := []struct {
```

Comment:
This can be simplified by removing some duplication:

```go
for _, n := range []int{1, 10, 100, 1000, 10000, 100000} {
  b.Run(fmt.Sprintf("Atomic/%v", n), func(...........
...
```

or

```go
for n := 1; n <= 100000; n *= 10 {
...
```